### PR TITLE
Rename DefaultPodTopologySpread plugin #91994

### LINKED ISSUE #91994: Renaming DefaultPodTopologySpread plugin
As suggested in https://github.com/kubernetes/kubernetes/pull/91793#issuecomment-642138709
This is necessary to avoid confusion between the plugin name and the feature gate that controls #80639

The name change should be fine as the plugin names were alpha, along with the component config API. And now that we are graduating to beta its the perfect time to change it.

One idea would be to call it `LegacyDefaultPodTopologySpread`, but since the replacement is only alpha, I don't think it's a good fit. I would suggest `DefaultPodSpread` instead (ie, remove Topology from it).

### PR DESCRIPTION
**Pull request by rakeshreddybandi**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
This PR renames the legacy DefaultPodTopologySpread plugin to SelectorSpread plugin.

**Which issue(s) this PR fixes**:

Fixes #91994

**Special notes for your reviewer**:
Based on discussion in these threads(https://github.com/kubernetes/kubernetes/issues/91994#issuecomment-643278479 and https://github.com/kubernetes/kubernetes/pull/91793#issuecomment-639659976), the legacy `DefaultPodTopologySpread` plugin is currently matching with feature name and so renamed it to `SelectorSpread` plugin

**Does this PR introduce a user-facing change?**:

```release-note
Renamed DefaultPodTopologySpread plugin to SelectorSpread plugin to avoid naming conflicts with feature Gate DefaultPodTopologySpread 
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by alculquicondor** [comment]
Please add a release note. This is user facing.

**Review by alculquicondor — COMMENTED** [review]
/ok-to-test

/approve

**Comment by alculquicondor** [comment]
It looks like you forgot to `./hack/update-bazel.sh`

**Comment by rakeshreddybandi** [comment]
Hi @alculquicondor Thanks for quick review.. 
Shoot I missed running this `./hack/update-bazel.sh` Let me fix it. Thanks

**Comment by rakeshreddybandi** [comment]
Test failure seems to be timeouts. Trying retest
/retest

File: CHANGELOG/CHANGELOG-1.19.md [hunk 1]
```diff
@@ -165,7 +165,7 @@ filename | sha512 hash
 
 - A new extension point `PostFilter` is introduced to scheduler framework which runs after Filter phase to resolve scheduling filter failures. A typical implementation is running preemption logic. ([#91314](https://github.com/kubernetes/kubernetes/pull/91314), [@Huang-Wei](https://github.com/Huang-Wei)) [SIG Scheduling and Testing]
 - Added --privileged flag to kubectl run ([#90569](https://github.com/kubernetes/kubernetes/pull/90569), [@brianpursley](https://github.com/brianpursley)) [SIG CLI]
-- Enable feature Gate DefaultPodTopologySpread to use PodTopologySpread plugin to do defaultspreading. In doing so, legacy DefaultPodTopologySpread plugin is disabled. ([#91793](https://github.com/kubernetes/kubernetes/pull/91793), [@alculquicondor](https://github.com/alculquicondor)) [SIG Scheduling]
+- Enable feature Gate DefaultPodTopologySpread to use PodTopologySpread plugin to do defaultspreading. In doing so, legacy DefaultPodTopologySpread plugin (Renamed to SelectorSpread plugin) is disabled. ([#91793](https://github.com/kubernetes/kubernetes/pull/91793), [@alculquicondor](https://github.com/alculquicondor)) [SIG Scheduling]
```
Comment (ahg-g) [inline CHANGELOG/CHANGELOG-1.19.md:168]: didn't know we can manually update change logs, is that really a thing?

Comment (alculquicondor) [reply CHANGELOG/CHANGELOG-1.19.md:168]: I have the same question

Comment (rakeshreddybandi) [reply CHANGELOG/CHANGELOG-1.19.md:168]: Sorry.. just observed the change log commit history and I dont see anyone updating it manually. Let me remove this change

Comment (alculquicondor) [reply CHANGELOG/CHANGELOG-1.19.md:168]: @justaugustus if we update a release note in a PR, would it eventually update the changelog?

**Comment by hprateek43** [comment]
Can you rebase this once? Changes look fine to me. Can add a `lgtm` once you rebase and tests pass.

**Comment by rakeshreddybandi** [comment]
Above failures doesn't look like changes related to this PR. Trying out retest
/retest

**Comment by alculquicondor** [comment]
@ahg-g anything to add?

**Comment by rakeshreddybandi** [comment]
Thank you @alculquicondor for helping on this PR

**Comment by alculquicondor** [comment]
Thank you for the contribution 😃

**Pull request merged**